kafka/producer.go

```diff
@@ -197,8 +209,10 @@ func (p *Producer) Close() error {
 // the messages have been stored in the producer's buffer.
 func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 	ctx, span := p.tracer.Start(ctx, "producer.ProcessBatch", trace.WithAttributes(
```

Comment:
It seems we're always starting a trace, even when `cfg.DisableTelemetry` is  set.

Reply:
Yeah, I setup `DisableTelemetry` thinking the kotel hook may have unintended side effects.
Disabling manual tracing can be done by not setting up an otel SDK though.

We could rename this to `DisableKgoTelemetry`, or entirely remove it?